runtime/lua/vim/lsp/util.lua

```diff
@@ -2135,6 +2135,41 @@ local function make_line_range_params(bufnr, start_line, end_line, position_enco
   }
 end
 
+---@class (private) vim.lsp.util._cancel_requests.Filter
+---@field bufnr? integer
+---@field clients? vim.lsp.Client[]
+---@field method? string
+---@field type? string
+
+---@private
+--- Cancel all {filter}ed requests.
+---
+---@param filter? vim.lsp.util._cancel_requests.Filter
+function M._cancel_requests(filter)
+  filter = filter or {}
+  local bufnr = filter.bufnr
```

Comment:
I could see a caller accidentally sending buf=0 expecting it to mean current buffer. So we should probably handle that.

```suggestion
  local bufnr = filter.bufnr and vim._resolve_bufnr(filter.bufnr) or nil
```